packages/web/src/components/RegistrationModal/RegistrationForm.tsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import {
+  Input,
+  FormControl,
+  VStack,
+  FormHelperText,
+  FormLabel,
+  Box,
+  Button,
+  Textarea,
+  InputProps,
+  TextareaProps,
+} from '@chakra-ui/react';
+import { AlertResponse } from './AlertResponse';
+import { useRegistrationConfig } from './useRegistrationConfig';
+import { RegistrationFormProps, FormProps, RegistrationSchema, RegistrationFormik } from './utils';
+import { Hint } from './HintTooltip';
+import { statusColors } from '../../theme/colors';
+
+const handleBlur = async (e: React.FocusEvent<HTMLInputElement>, formik: RegistrationFormik) => {
+  const values = {
+    [e.target.id]: e.target.value,
+    // if any outstanding validation errors exist, add them to be validated
+    ...Object.entries(formik.errors)
+      .map(([key]) => {
+        const value = formik.values[key as keyof RegistrationSchema];
+        return { [key]: value };
+      })
+      .reduce((acc, curr) => ({ ...acc, ...curr }), {}),
+  };
+  await formik.validateForm(values);
+};
+const formProps: FormProps = (key, formik) => ({
+  variant: 'filled',
+  type: 'text',
+  value: formik.values[key],
+  isInvalid: !!formik.errors[key],
+  onChange: formik.handleChange,
+  onBlur: (e: React.FocusEvent<HTMLInputElement>) => handleBlur(e, formik),
+});
```

Comment:
Try this for your types issues
```suggestion
} as const);
```